### DISCUSSION TITLE
NextJS building failure

### DISCUSSION THREAD

**Discussion by zowlou** [vercel/next.js#74027] [Help]
### Summary

Hi everyone, 

I'm building an app in NextJS + React. 
When I work on it in dev, I don't have any problems. 
But when I want to build it, it doesn't work. 

I've already deleted node_modules and .next, but nothing has changed. 

Here's the error feedback I get: 
`$ npm run build

> Pharma Manufacturing@2.0.0 build
> next build

  ▲ Next.js 14.2.20

   Creating an optimized production build ...
 ✓ Compiled successfully
   Linting and checking validity of types  ..Failed to compile.

.next/types/app/quality/page.ts:28:13
Type error: Type 'OmitWithTag<Readonly<BenefitsProps>, keyof PageProps, "default">' does not satisfy the constraint '{ [x: string]: never; }'.
  Property 'data' is incompatible with index signature.
    Type 'Data' is not assignable to type 'never'.

  26 |
  27 | // Check the prop type of the entry function
> 28 | checkFields<Diff<PageProps, FirstArg<TEntry['default']>, 'default'>>()
     |             ^
  29 |
  30 | // Check the arguments and return type of the generateMetadata function
  31 | if ('generateMetadata' in entry) {`

I don't know what elements you need to see the problems. 

![scheme](https://github.com/user-attachments/assets/a3a9a903-a2bd-407b-a5bc-b7f55e609672)

Thank you for your help.

**Comment by hyjoong** [comment]
Here's the error you're seeing is a TypeScript type error that occurs during build time. To help you solve this issue, I need to see your quality/page.tsx file.
(Optional) If the issue persists, sharing your tsconfig.json might also be helpful.

**Reply by zowlou** [reply]
Here's the quality/page.tsx

`import { Container } from '@/components/Container'
import { StaticImageData } from 'next/image';

import Certification from '@/components/Certification'

import benefitTwoImg from "../../../public/img/benefit-two.png";


interface Bullet {
  title: string;
  desc: string;
  icon: React.ReactNode;
}

interface Data {
    imgPos?: "left" | "right";
    title: string;
    desc: string;
    desc2?: string | JSX.Element; // Utiliser JSX.Element si c'est un composant React
    desc3?: string | JSX.Element;
    image: string | StaticImageData ; // Utilisez `string` ou `StaticImageData` pour les images
    image2?: string | StaticImageData;
    image3?: string | StaticImageData ;
    bullet?: boolean;
    bullets:Bullet[];
}

interface BenefitsProps {
  imgPos?: "left" | "right";
  data: Data;
}

const data = {
    title: "Qualité",
    desc: "Contrôle de qualité permanent par notre pharmacien d’industrie, notre pharmacien et nos deux diététiciens :",
    image: benefitTwoImg,

}

const Quality = (props: Readonly<BenefitsProps>) => {

    return (
        <Container className="flex flex-wrap flex-col-reverse mb-20 lg:gap-10 lg:flex-nowrap lg:flex-row">
            <div
                className={`flex items-center justify-center w-full lg:w-1/2 ${props.imgPos === "right" ? "lg:order-1" : ""
                    }`}>
                <div>
                    <Certification />
                </div>
            </div>

            <div
                className={`flex flex-wrap items-center w-full lg:w-1/2`}>
                <div>
                    <div className="flex flex-col w-full mt-4">
                        <h3 className="max-w-2xl mt-3 text-3xl font-bold leading-snug tracking-tight text-gray-800 lg:leading-tight lg:text-4xl dark:text-white">
                            {data.title}
                        </h3>
                        <p className="max-w-2xl py-4 text-lg leading-normal text-gray-500 lg:text-xl xl:text-xl dark:text-gray-300 white-space: pre-wrap">
                            {data.desc}
                        </p>
                        <div className="w-full mt-5">
                            <div className="flex items-start mt-8 space-x-3">
                                <ul className="mt-1 text-gray-500 dark:text-gray-400 list-disc ">
                                    <li></li>
                                    <li></li>
                                    <li></li>
                                    <li></li>
                                </ul>
                            </div>
                        </div>
                    </div>


                </div>
            </div>
        </Container>
    )
}

export default Quality`


And the optional tsconfig.json
`{
  "compilerOptions": {
    "lib": ["dom", "dom.iterable", "esnext"],
    "allowJs": true,
    "skipLibCheck": true,
    "strict": true,
    "noEmit": true,
    "esModuleInterop": true,
    "module": "esnext",
    "moduleResolution": "bundler",
    "resolveJsonModule": true,
    "isolatedModules": true,
    "jsx": "preserve",
    "incremental": true,
    "plugins": [
      {
        "name": "next"
      }
    ],
    "paths": {
      "@/*": ["./src/*"]
    }
  },
  "include": ["next-env.d.ts", "**/*.ts", "**/*.tsx", ".next/types/**/*.ts", "src/components/data.txs", "src/components/data.js"],
  "exclude": ["node_modules"]
}
`

**Reply by icyJoseph** [reply]
The problem is that `Quality` takes in props that do not match what `page.tsx` takes, which are `params`, and `searchParams`, as promises. 

Create a `quality.tsx` file, and move your contents there, and then in your `page.tsx` file:

```jsx
import Quality from './quality';

export default function Page() {
   return <Quality />
}
```

And fix other imports of Quality in your app.
